internal/core/meta/meta.go

```diff
@@ -6,6 +6,7 @@ package meta
 // access.
 // TODO: get rid of this since it's hacky
 type AgentMeta struct {
-	InternalStatusHost string
-	InternalStatusPort uint16
+	InternalStatusHost       string
+	InternalStatusPort       uint16
+	EnableExtraMetricsFilter bool
```

Comment:
As the comment on this indicates, I don't really want to expand the use of this.  If this is a permanent thing and not just a transitional hack, I would rather find some different to way to do this.

The main problem with AgentMeta as it stands now is that changes to it don't necessarily get observed by monitors, nor is there any notification system for it, so the only robust way to ensure monitors are in sync with this config is to restart all of them if something in AgentMeta gets changed, or add some notification callback system to it.